sys/net/application_layer/asymcute/asymcute.c

```diff
@@ -83,13 +83,15 @@ static size_t _len_set(uint8_t *buf, size_t len)
     }
 }
 
-static size_t _len_get(uint8_t *buf, size_t *len)
+static ssize_t _len_get(uint8_t *buf, size_t pkt_len, size_t *len)
 {
     if (buf[0] != 0x01) {
         *len = (uint16_t)buf[0];
         return 1;
     }
     else {
+        if (pkt_len <= 3)
```

Comment:
Why `<=` and not `<`? From what I gather of the code below, it tries to read a 2 byte buffer from position 1 of the packet, so `pkt_len == 3` should be legal and for anything packet validity related, there is [a check in `_on_pkt()`](https://github.com/RIOT-OS/RIOT/pull/18433/files#diff-d63d8a165e6b370754ebb39bfeca42f1ed38f7c8a53a59fae1cb4db14fc7d8e6R590)

Reply:
My thinking was just that the message needs data apart from the length header, but you are right. This is caught by the check in `_on_pkt()` and hence `<` should be sufficient. Thanks for pointing this out, I adjusted the check accordingly.